mlflow/data/delta_dataset_source.py

```diff
@@ -86,6 +109,27 @@ def _is_databricks_uc_table(self):
                 and catalog_name != DATABRICKS_SAMPLES_CATALOG_NAME
             )
 
+    def _lookup_table_id(self, table_name):
+        try:
+            req_body = message_to_json(GetTable(full_name_arg=table_name))
+            _METHOD_TO_INFO = extract_api_info_for_service(
+                UnityCatalogService, _REST_API_PATH_PREFIX
+            )
+            endpoint, method = _METHOD_TO_INFO[GetTable]
+            table_info: TableInfo = call_endpoint(
+                host_creds=functools.partial(
+                    get_databricks_host_creds, _DATABRICKS_UNITY_CATALOG_SCHEME
+                ),
+                endpoint=endpoint,
+                method=method,
+                json_body=req_body,
+                response_proto=GetTable.Response,
+            ).table_info
+            return table_info.table_id
+        except Exception as e:
+            _logger.warning(f"Unable to look up the UC table id for table '{table_name}': '", e)
```

Comment:
not familiar with the API request at hand, but do we want to add more descriptive error handling (ie. authentication error, invalid request, etc.)? maybe this is already taken care of with `rest_utils`

Reply:
Sid suggests to get rid of this logging altogether, in an earlier comment.